### PR TITLE
bazel: refactor openapi-gen logic for new kazel

### PR DESCRIPTION
**Pull request by ixdy**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: I recently refactored the kubernetes codegen logic in kazel to support additional code generators beyond just openapi-gen.

This PR refactors the openapi code generation under to use the new starlark dictionaries produced by kazel.

There's no real change in functionality, though adding/removing openapi packages will now be reflected in `build/kazel_generated.bzl` instead of `pkg/generated/openapi/BUILD`.

I've also updated the import of `kazel` from `github.com/kubernetes/repo-infra/kazel` to `k8s.io/repo-infra/kazel`.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/hold
because it needs to revendor kazel

cc @fejta @BenTheElder

### CODE REVIEW

**Comment by ixdy** [comment]
ping?

File: build/code_generation.bzl
Comment (BenTheElder) [inline build/code_generation.bzl]: does this work? seems like this should be two arguments? `"--v", "1"` or `--v=1`

Comment (BenTheElder) [reply build/code_generation.bzl]: (there are a lot of these so I assume it works, but you usually only see this when it's either bugged, or it's being passed to be evaluated in a shell)

**Review by BenTheElder — APPROVED** [review]
/lgtm
/hold
looks fine to me, would like @fejta's review as I think he's written more bazel rules / macros than I have.

File: build/code_generation.bzl
Comment (ixdy) [reply build/code_generation.bzl]: this is moved from the preexisting pkg/generated/openapi/def.bzl.
these all get joined as a single space-separated string anyway - look at L66.

Comment (BenTheElder) [reply build/code_generation.bzl]: :+1: yay space separated arg lists! /s

File: build/BUILD
Comment (fejta) [inline build/BUILD]: I would recommend formatting this as:

```python
code_generation_test_suite(name = "code_generation_tests")
```

to help with autocomplete (and then piping this into the macro)

File: build/code_generation_test.bzl
Comment (fejta) [inline build/code_generation_test.bzl]: `name` if you take the previous suggestion

**Review by fejta — COMMENTED** [review]
/lgtm

**Comment by fejta** [comment]
I don't see `//build:code_generation_tests` in https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/71529/pull-kubernetes-bazel-test/69013/?log#log, shouldn't I?

**Comment by ixdy** [comment]
Good catch.
https://github.com/kubernetes/kubernetes/blob/2bd14ea60e934f0d7a499a1a830a05262238605a/build/root/Makefile#L570-L578
(now trying to think about the best way to address this)

File: build/BUILD
Comment (ixdy) [reply build/BUILD]: good suggestion, done

**Comment by ixdy** [comment]
ok, this includes a re-vendoring of kazel, so hopefully is ready to go.
/hold cancel
/assign @thockin 
for approval

**Comment by fejta** [comment]
Still not seeing `//build: code_generation_tests` in https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/71529/pull-kubernetes-bazel-test/69441/?

**Comment by ixdy** [comment]
oh, yeah, because we don't actually use `make bazel-test` on prow:
https://github.com/kubernetes/test-infra/blob/8fed8ae80a92784c1eee0abeb8115aa1b0f52f09/config/jobs/kubernetes/sig-testing/bazel-build-test.yaml#L208

**Comment by BenTheElder** [comment]
> oh, yeah, because we don't actually use make bazel-test on prow:

maybe we should get around to changing that? do we still want to?

**Comment by thockin** [comment]
Godeps, etc LGTM

/approve

**Comment by ixdy** [comment]
rebased, need re-lgtm

**Pull request merged**
